node/rest/src/helpers/auth.rs

```diff
@@ -67,28 +72,29 @@ impl Claims {
     }
 }
 
-/// Checks the authorization header for a valid token.
-pub fn with_auth() -> impl Filter<Extract = ((),), Error = Rejection> + Clone {
-    warp::header::<String>("authorization").and_then(|token: String| async move {
-        if !token.starts_with("Bearer ") {
-            return Err(reject::custom(RestError::Request("Invalid authorization header.".to_string())));
+pub async fn auth_middleware<B>(request: Request<B>, next: Next<B>) -> Result<Response, StatusCode>
+where
+    B: Send,
+{
+    // Deconstruct the request to extract the auth token.
+    let (mut parts, body) = request.into_parts();
+    let auth: TypedHeader<Authorization<Bearer>> = parts.extract().await.map_err(|_| StatusCode::UNAUTHORIZED)?;
+
+    match decode::<Claims>(auth.token(), &DecodingKey::from_secret(jwt_secret()), &Validation::new(Algorithm::HS256)) {
+        Ok(decoded) => {
+            let claims = decoded.claims;
+            if claims.is_expired() {
+                return Err(StatusCode::UNAUTHORIZED);
+            }
         }
 
-        // Decode the claims from the token.
-        match decode::<Claims>(
-            token.trim_start_matches("Bearer "),
-            &DecodingKey::from_secret(jwt_secret()),
-            &Validation::new(Algorithm::HS256),
-        ) {
-            Ok(decoded) => {
-                let claims = decoded.claims;
-                if claims.is_expired() {
-                    return Err(reject::custom(RestError::Request("Expired JSON Web Token.".to_string())));
```

Comment:
The old code passed a message to the caller. Is that not possible, or just to minimize the info for security reasons?

Reply:
Sure, this should be possible; now it just returns a 401 which is reasonable for an expired JWT. I don't think we need to worry too much about minimising info so I can take a look at adding this detail back in 👍

Reply:
note: it's currently an extremely unlikely error to encounter because the expiry is set to 10 years 😄

Reply:
done